lib/generators/datepicker/datepicker_generator.rb

```diff
@@ -0,0 +1,68 @@
+require 'rails/generators/base'
+
+class DatepickerGenerator < Rails::Generators::Base
+
+  source_root File.expand_path('../templates', __FILE__)
+
+  def modify_application_js
+    js_file
+    assets = ['datepicker', 'jquery.mask']
+    assets.each do |asset|
+      if File.open(js_root_path + js_file).grep(/(\/\/= require #{asset})/).size > 0
+        puts "//= require #{asset} already exists in #{js_file}..."
+      else
+        open(js_root_path + js_file, 'a') do |file|
+          file << "//= require #{asset}\n"
+        end
+        puts "//= require #{asset} added to #{js_file}..."
+      end
+    end
+  end
+
+  def modify_application_css
+    style_file
+    text = File.read(style_root_path + style_file) unless style_file.nil?
+    if File.readlines(style_root_path + style_file).grep(/(\*\= require datepicker)/).size > 0
+      puts "*= require datepicker already exists in #{style_file}..."
+    else
+      if File.readlines(style_root_path + style_file).grep(/(\*\= require_tree \.)/).size > 0
+        new_file_contents = text.gsub(/(\*\= require_tree \.)/, "*= require datepicker\n *= require_tree .")
+      else
+        new_file_contents = text.gsub(/(\*\/)/, " *= require datepicker\n*/")
+      end
+      File.open(style_root_path + style_file, "w") { |file| file.puts new_file_contents }
+      puts "*= require datepicker added to #{style_file}..."
+    end
+  end
+
+  def finish
+    puts "Done."
+  end
+
+  private
+
+  def style_file
+    css = 'application.css'
+    scss = 'application.scss'
+    if File.file?(style_root_path + css)
```

Comment:
You can cut the line count down with this approach

```ruby
def style_file
  css = 'application.css'
  scss = 'application.scss'
  return css if File.file?(style_root_path + css)
  File.file?(style_root_path + scss) ? scss : nil
end
```

Some additional thoughts.  I'd probably use a variable name `css_file_name` over `css`.  `css` implies the variable holds some raw CSS.  It's not a big deal in this method because the method is so short.   If this were a parameter to a method, it would be better to have a more explicit variable name, because you can't actually see the value of the variable (like you can in this method).  If I saw a method like ` def foo(css)`, I wouldn't assume the `css` param was actually a file name. `def foo(css_file_name)` makes it clear.